gennav/envs/binaryOccupancyGrid2D_env.py

```diff
@@ -0,0 +1,199 @@
+import numpy as np
+from gennav.envs.base import Environment
+from gennav.utils.common import RobotState
+from gennav.utils.geometry import Point
+from matplotlib import pyplot as plt
+
+
+class BinaryOccupancyGrid2DEnv(Environment):
+    """Base class for a Binary Occupancy Grid 2D envrionment.
+
+        Arguments:
+            X (unsigned int) : The number of grid cells in the x-direction
+            Y (unsigned int) : the number of grid cells in the y-direction
+    """
+
+    def __init__(self, X=10, Y=10):
+        super(BinaryOccupancyGrid2DEnv, self).__init__()
+        self.X = X
+        self.Y = Y
+        self.scan = None
+        self.robotPose = None
+        self.scale = 5
+        self.grid = np.zeros((self.X * self.scale, self.Y * self.scale))
+
+        # Storing transforms
+        self.transforms = {}
+        self.mapTbot = {
+            "from": "map",
+            "to": "bot",
+            "transform": self.scale
+            * np.array(
+                [[1, 0, int(self.X / 2)], [0, 1, int(self.Y / 2)], [0, 0, 1]]
+            ).reshape(3, 3),
+        }
+        self.botTworld = {"from": "bot", "to": "world", "transform": np.empty((3, 3))}
+        self.mapTworld = {
+            "from": "map",
+            "to": "world",
+            "transform": np.dot(self.mapTbot["transform"], self.botTworld["transform"]),
+        }
+        self.transforms["mapTbot"] = self.mapTbot
+        self.transforms["botTworld"] = self.botTworld
+        self.transforms["mapTworld"] = self.mapTworld
+
+    def update(self, scan, robotPose):
```

Comment:
You should only updated with `scan` here. Why is the `robotPose` argument necessary?

Reply:
The robotPose argument is necessary to construct transform between the bot frame and the world frame, which is required to compute a transform from the map frame to the world which will be used in trajectory validity or pose validity.